Problem3.java

```diff
@@ -0,0 +1,40 @@
+// Time Complexity : O(log(n))
+// Space Complexity :O(1)
+// Did this code successfully run on Leetcode : yes
+// Any problem you faced while coding this : Did not understand what does it mean when the question says, nums[-1] == nums[n] == -Infinity ???
```

Comment:
Its just an assumption. Just consider the -inf is before the first element i.e nums[-1] and after last element i.e nums[n]